lib/connect.ts

```diff
@@ -65,1354 +69,648 @@ import utils from "./utils";
-        return request(route, "POST", params || {}, responseType, responseTransformer, isJSON, queryParams);
-    }
+    return data;
+}
 
-    function _put(route, params, responseType?, responseTransformer?, isJSON = false, queryParams=null) {
-        return request(route, "PUT", params || {}, responseType, responseTransformer, isJSON, queryParams);
-    }
+function parseHistorical(jsonData: AnyObject): AnyObject {
+    // Return if its an error
+    if (jsonData.error_type) return jsonData;
+
+    const results = [] as any[];
+    for (let i = 0; i < jsonData.data.candles.length; i++) {
+        const d = jsonData.data.candles[i];
+        const c: AnyObject = {
+            'date': new Date(d[0]),
+            'open': d[1],
+            'high': d[2],
+            'low': d[3],
+            'close': d[4],
+            'volume': d[5]
+        }
+
+        // Add OI field if its returned
+        if (d[6]) {
+            c['oi'] = d[6]
+        }
 
-    function _delete(route, params, responseType?, responseTransformer?, isJSON = false) {
-        return request(route, "DELETE", params || {}, responseType, responseTransformer, isJSON);
+        results.push(c);
     }
 
-    function request(route, method, params, responseType?, responseTransformer?, isJSON?, queryParams?) {
-        // Check access token
-        if (!responseType) responseType = "json";
-        let uri = routes[route];
+    return { 'data': results };
+}
 
-        // Replace variables in "RESTful" URLs with corresponding params
-        if(uri.indexOf("{") !== -1) {
-            let k;
-            for(k in params) {
-                if(params.hasOwnProperty(k)) {
-                    uri = uri.replace("{" + k + "}", params[k]);
-                }
+function transformInstrumentsResponse(data: any, headers: AnyObject): unknown[] | any {
+    // Parse CSV responses
+    if (headers['content-type'] === 'text/csv') {
+        const parsedData = csvParse.parse(data, { 'header': true }).data;
+        for (const item of parsedData as any) {
+            item['last_price'] = parseFloat(item['last_price']);
+            item['strike'] = parseFloat(item['strike']);
+            item['tick_size'] = parseFloat(item['tick_size']);
+            item['lot_size'] = parseInt(item['lot_size']);
+
+            if (item['expiry'] && item['expiry'].length === 10) {
+                item['expiry'] = new Date(item['expiry']);
             }
         }
 
-        let payload: null | string = null;
-        if (method === "GET" || method === "DELETE") {
-            queryParams = params;
-        } else {
-            if (isJSON) {
-                // post JSON payload
-                payload = JSON.stringify(params);
-            } else {
-                // post url encoded payload
-                payload = querystring.stringify(params);
+        return parsedData;
+    }
+
+    return data;
+}
+
+function transformMFInstrumentsResponse(data: any, headers: AnyObject): any {
```

Comment:
It is advisable to designate the `transformMFInstrumentsResponse` function as a private function.